source/codegen/templates/service_helpers.mako

```diff
@@ -309,7 +309,7 @@ ${initialize_standard_input_param(function_name, parameter)}\
 %     elif service_helpers.is_output_array_that_needs_coercion(parameter):
       std::vector<${underlying_param_type}> ${parameter_name}(${size});
 ## uInt32 requires cast because of int vs long in that typedef vs uint32_t
-%     elif underlying_param_type in ['ViAddr', 'ViInt32', 'ViUInt32', 'ViUInt16', 'uInt32']:
+%     elif underlying_param_type in ['ViAddr', 'ViInt32', 'ViUInt32', 'ViUInt16', 'uInt32', 'int32']:
```

Comment:
More casts required because of different definitions of 32-bit integers (int vs long)